### PR TITLE
Feature/ignore data outside region of interest

### LINKED ISSUE #38: `remove_restore.py` overall docstring is out-of-date
We should update the `remove_restore.py` docstring and add a docstring to `load_base_grid` too. We might want to expand/review the docs when we tackle this, too.

### PR DESCRIPTION
**Pull request by JamieJQuinn**
This PR adds the ability to ignore any update grids which lie entirely outwith the region of interest.

It also fixes #38

### CODE REVIEW

File: pycascadia/remove_restore.py [hunk 1]
```diff
@@ -1,8 +1,7 @@
 #!/usr/bin/env python3
 
 """
-This script should not be used in any serious way (in its current form at least)!
-It's just a way for us to learn how to use pyGMT and how to read/write the sample data.
+Implementation of the remove-restore algorithm from the GEBCO cookbook
```
Comment (alessandrofelder) [inline pycascadia/remove_restore.py:4]: Maybe specify?
```suggestion
Implementation of the remove-restore algorithm from the GEBCO cookbook

Note that the remove-restore algorithm consists of only "step D" described in the cookbook. Preprocessing steps A-C are not included here.
```

File: pycascadia/remove_restore.py [hunk 2]
```diff
@@ -13,13 +12,17 @@
...
-from pycascadia.utility import region_to_str, min_regions
+from pycascadia.utility import region_to_str, min_regions, is_region_valid
 
 def calc_diff_grid(base_grid, update_grid, diff_threshold=0.0, window_width=None):
     """Calculates difference grid for use in remove-restore"""
     print("Blockmedian update grid")
     max_spacing = max(update_grid.spacing, base_grid.spacing)
     minimal_region = min_regions(update_grid.region, base_grid.region)
+    if not is_region_valid(minimal_region):
+        print("Update grid is outside region of interest. Skipping.")
```
Comment (alessandrofelder) [inline pycascadia/remove_restore.py:23]: If I understand correctly, we could make this clearer by
```suggestion
        print("Update grid is (at least partially) outside region of interest. Skipping.")
```

**Review by alessandrofelder — APPROVED** [review]
Awesome, thanks!
Some minor, nitpicky documentation-related suggestions - feel free to ignore if you disagree.

File: pycascadia/remove_restore.py (hunk 2 above)
Comment (JamieJQuinn) [reply pycascadia/remove_restore.py:25]: If it's only partially outside the ROI its data will be included. This check is only for when the update grid is *entirely* outside the ROI and can be skipped.

Comment (alessandrofelder) [reply pycascadia/remove_restore.py:25]: Right, yes, OK, agreed 🤦 
Then maybe? 😆 
```suggestion
        print("Update grid is entirely outside region of interest. Skipping.")
```

**Pull request merged**